File: evidence/generated-systems/golang/insurance-on-golang/insurance-on-golang/internal/model/Policy.go
package model

import (
    "gorm.io/gorm"
)

//==============================================================
// Policy Declaration
//==============================================================
type Policy struct {
    gorm.Model
     PolicyNumber                                                            string
    EffectivePeriod                                                            string
    TotalPremium                                                            string
    InsurerId         *uint
    Insurer           *Insurer `gorm:"foreignKey:InsurerId"`
    CustomerId         *uint
    Customer           *Customer `gorm:"foreignKey:CustomerId"`
    ProductId         *uint
    Product           *InsuranceProduct `gorm:"foreignKey:ProductId"`
    AgentId         *uint
    Agent           *Agent `gorm:"foreignKey:AgentId"`
     Coverages           []PolicyCoverage `gorm:"foreignKey:CoveragesFromPolicyId"`
     InsuredObjects           []InsuredObject `gorm:"foreignKey:InsuredObjectsFromPolicyId"`
     Endorsements           []Endorsement `gorm:"foreignKey:EndorsementsFromPolicyId"`
    BillingAccountId         *uint
    BillingAccount           *BillingAccount `gorm:"foreignKey:BillingAccountId"`
     Beneficiaries           []Beneficiary `gorm:"foreignKey:BeneficiariesFromPolicyId"`
     Claims           []Claim `gorm:"foreignKey:ClaimsFromPolicyId"`
     ReinsuranceAgreements           []ReinsuranceAgreement `gorm:"foreignKey:ReinsuranceAgreementsFromPolicyId"`
    Status                      PolicyStatus
    PaymentPlan                      PaymentPlanType

// parent associations as their child

}

File: evidence/generated-systems/rubyonrails/insurance-on-rails/insuranceonrails/app/models/policy.rb
class Policy < ApplicationRecord
  enum Status: [:Quoted, :Active, :Lapsed, :Cancelled, :Expired, :PendingCancel, :PendingReinstatement]
  enum PaymentPlan: [:Annual, :SemiAnnual, :Quarterly, :Monthly, :PayInFull]


  composed_of :policyNumber,
    class_name: "PolicyNumber",
    mapping: [
      %w[policyNumber_value value]
    ]

  composed_of :dateRange,
    class_name: "DateRange",
    mapping: [
      ${$mapping}, 
      %w[dateRange_endDate endDate]
    ]

  composed_of :money,
    class_name: "Money",
    mapping: [
      ${$mapping}, 
      %w[money_currency currency]
    ]

  has_many :Insurer, class_name: 'Insurer'
  has_many :Customer, class_name: 'Customer'
  has_many :Product, class_name: 'InsuranceProduct'
  has_many :Agent, class_name: 'Agent'
  has_many :Coverages, class_name: 'PolicyCoverage'
  has_many :InsuredObjects, class_name: 'InsuredObject'
  has_many :Endorsements, class_name: 'Endorsement'
  has_many :BillingAccount, class_name: 'BillingAccount'
  has_many :Beneficiaries, class_name: 'Beneficiary'
  has_many :Claims, class_name: 'Claim'
  has_many :ReinsuranceAgreements, class_name: 'ReinsuranceAgreement'

end
